adafruit_tca8418.py

```diff
@@ -343,7 +355,7 @@ def direction(self, val):
         self._dir = val
 
     @property
-    def pull(self):
+    def pull(self) -> Union[None, Literal[digitalio.Pull.UP]]:
```

Comment:
In place of the `Union`, we should use `Optional[Literal[digitalio.Pull.UP]]`.

Reply:
haven't we done it both ways?  I thought it was preferable to avoid Optional on returns - so do you have any wisdom or reference to pass on regarding this?

Reply:
Son in later versions of Python it gets simplified a lot:  You can actually use `|` instead of `Union`, in which case you would just use `None`:

```python
def sometime_str(give_str: bool) -> str | None:
    if give_str:
        return "Here's a string!"
    else:
        return None
```

Unfortunately, that isn't supported in Python 3.7, which is the minimum version that Blinka supports, so instead it should be `Optional[str]`.  But technically `Union[..., None]` is equivalent to `Optional[...]`, using Optional is just the shorthand for it.